pkg/services/cloudmigration/cloudmigrationimpl/cloudmigration.go

```diff
@@ -532,19 +534,43 @@ func (s *Service) GetSnapshot(ctx context.Context, query cloudmigration.GetSnaps
 		// ask GMS for status if it's in the cloud
 		snapshotMeta, err := s.gmsClient.GetSnapshotStatus(ctx, *session, *snapshot)
 		if err != nil {
-			return nil, fmt.Errorf("error fetching snapshot status from GMS: sessionUid: %s, snapshotUid: %s", sessionUid, snapshotUid)
+			return snapshot, fmt.Errorf("error fetching snapshot status from GMS: sessionUid: %s, snapshotUid: %s", sessionUid, snapshotUid)
+		}
+
+		var localStatus cloudmigration.SnapshotStatus
+		switch snapshotMeta.State {
```

Comment:
This is fine, but could it be just a map `Status -> State`?

Reply:
Done